tests/pytest_framework/src/syslog_ng/syslog_ng_cli.py

```diff
@@ -54,30 +54,51 @@ def __syntax_only(self, config_path=None):
             command_short_name="syntax_only", command=["--syntax-only", "--cfgfile={}".format(config_path)]
         )
 
-    # Process commands
-    def start(self, config):
-        self.__logger.info("Beginning of syslog-ng start")
-        config.write_config_content()
-
-        # syntax check
+    def __syntax_check(self):
         result = self.__syntax_only()
         if result["exit_code"] != 0:
             self.__logger.error(result["stderr"])
             raise Exception("syslog-ng can not started")
 
-        # effective start
-        if self.__valgrind_usage:
-            self.__process = self.__syslog_ng_executor.run_process_with_valgrind()
-        else:
-            self.__process = self.__syslog_ng_executor.run_process()
-
+    def __is_process_running(self):
+        return self.__process.poll() == None
+
+    def wait_for_control_socket_alive(self):
+        class StoppedUnexpectedly(Exception):
+            pass
+        def is_alive(s):
+            if not s.__is_process_running():
+                raise StoppedUnexpectedly
+            return s.__syslog_ng_ctl.is_control_socket_alive()
+        try:
+            return wait_until_true(is_alive, self)
+        except StoppedUnexpectedly:
+            return False
+
+    def wait_for_start(self):
         # wait for start and check start result
-        if not self.__syslog_ng_ctl.wait_for_control_socket_alive():
+        if not self.wait_for_control_socket_alive():
```

Comment:
This will not needed

Reply:
As mentioned above, wait_for_process_alive cannot help us now.